web3/contract.py

```diff
@@ -1328,6 +1332,39 @@ def factory(cls, class_name, **kwargs):
         return PropertyCheckingFactory(class_name, (cls,), kwargs)
 
 
+class CallerMethod:
+    def __init__(self, function, normalizers=None):
+        self._function = function
+
+    def __call__(self, *args, **kwargs):
+        # TODO: Remove this and consolidate into ContractCaller.
```

Comment:
Maybe you'd want it to be a staticmethod in ContractCaller or something. Or it could make sense to keep it as a separate method (doesn't need to be a class). Something like:

```py
def call_contract_function(fn, *args, transaction_dict=None, **kwargs):
    if transaction_dict is None:
        transaction_dict = {}
    return fn(*args, **kwargs).call(transaction_dict)
```

Then later in the `ContractCaller`, something like:
```py
for func in self._functions:
    ...
    caller_method = functools.partial(call_contract_function, fn, transaction_dict=transaction_dict)